docs/docs/view-components.md

```diff
@@ -149,49 +160,73 @@ Now implement `NitroImageViewManager` in Swift and Kotlin, and assume it has to
   <TabItem value="swift" label="iOS (Swift)" default>
     ```swift
     class HybridNitroImageViewManager: HybridNitroImageViewManagerSpec {
-      private let view: NitroImageView
-      init(withView view: NitroImageView) {
-        self.view = view
+      public var hybridContext = margelo.nitro.HybridContext()
+      public var memorySize: Int {
+        return getSizeOf(self)
+      }
+      private var nitroId: Double? = nil
+      private weak var view: NitroImageView? {
+        get {
+            guard let viewId = self.nitroId else {
+                return nil
+            }
+            return NitroImageView.globalViewsMap.object(forKey: NSNumber(value: viewId))
+        }
       }
 
       var image: Image {
         get { return view.image }
         set { view.image = newValue }
       }
+
       var opacity: Double {
         get { return view.opacity }
         set { view.opacity = newValue }
       }
+
+      func setNitroId(nitroId: Double) {
+        self.nitroId = nitroId
+      }
     }
     ```
+
   </TabItem>
   <TabItem value="kotlin" label="Android (Kotlin)">
     ```kotlin
     class HybridNitroImageViewManager: HybridNitroImageViewManagerSpec() {
-      private val view: NitroImageView
-      constructor(view: NitroImageView) {
-        this.view = view
-      }
+      private var nitroId: Double? = null
+
+      private var view: WeakReference<NitroImageView>? = null
+        get() {
+          return nitroId.let {
+            return NitroImageView.globalViewsMap[it]?.get()
+          }
+        }
```

Comment:
Why did you change that? To make it weak, or what?

Reply:
Yes, I make it weak and nothing stored at HybridNitroImageViewManager exclude nitroId.
On prev code:
```kotlin
      private val view: NitroImageView
      constructor(view: NitroImageView) {
        this.view = view
      }
```
u stored strong view to variable. it will be exist when own view destroy by react. Similar for iOS, like this:
```tsx
const App = () => {
  const [shown, setShown] = useState(true);

  const toggle = () => {
    setShown((v) => !v);
  };

  return (
    <View>
      <Button title="Toggle" onPress={toggle} />
      {shown ? <NitroImage nitroId={123} /> : null}
    </View>
  );
};
```
One more thing, when run **createViewManager(nitroId)** on didMount. it can running before nitroId props change of the native view. I tested with RN 0.75.
so by this way, will return null:
```swift
        let view = NitroImage.globalViewsMap.object(forKey: nitroId)
        return HybridNitroImageViewManager(withView: view)
```

Reply:
And when i use autolinking. I can't find contructor parameter "withView". So i can't run my code with prev guideline

Reply:
Got it! thanks